src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -81,7 +81,10 @@ public override async ValueTask HandleRpc(ClientPlayer sender, ClientPlayer? tar
                     }
 
                     var taskId = reader.ReadPackedUInt32();
-                    await _eventManager.CallAsync(new PlayerCompletedTaskEvent(_game, sender, this, taskId));
+                    var task = PlayerInfo.Tasks[(int)taskId] ??= new InnerGameData.TaskInfo();
```

Comment:
Maybe throw a warning if the task doesn't exist?

Reply:
I will throw a warning if the player tries to complete a task they don't have or complete a task they have already done.